redisdb/datadog_checks/redisdb/redisdb.py

```diff
@@ -282,8 +282,12 @@ def _check_db(self):
                 # client_list is disabled on some environments
                 self.log.debug("Unable to collect client metrics: CLIENT disabled in some managed Redis.")
 
-        # Save the number of commands.
-        self.rate('redis.net.commands', info['total_commands_processed'], tags=tags)
+        # Avoid corner case error by ensuring availability in info before collecting
+        if 'total_commands_processed' in info:
+            # Save the number of commands.
+            self.rate('redis.net.commands', info['total_commands_processed'], tags=tags)
+        else:
+            self.log.debug("total_commands_processed not found in info, skipping. Info: %s", info)
```

Comment:
I wonder if we should emit 0 in that case for the `redis.net.commands` metric?

Reply:
As we are not even receiving this data, not sending the metric would be the expected behavior.